src/lib.rs

```diff
@@ -463,41 +452,289 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> HotShot<TYPES::ConsensusType
+                    consensus.metrics.outstanding_transactions.update(1);
+                    consensus
+                        .metrics
+                        .outstanding_transactions_memory_size
+                        .update(i64::try_from(size).unwrap_or(i64::MAX));
+                }
+            }
+        }
+    }
+
+    /// Handle an incoming [`DataMessage`] that directed at this node
+    #[allow(clippy::unused_async)] // async for API compatibility reasons
+    async fn handle_direct_data_message(
+        &self,
+        msg: DataMessage<TYPES>,
+        _sender: TYPES::SignatureKey,
+    ) {
+        debug!(?msg, "Incoming direct data message");
+        match msg {
+            DataMessage::SubmitTransaction(_, _) => {
+                // Log exceptional situation and proceed
+                warn!(?msg, "Broadcast message received over direct channel");
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl<
+        TYPES: NodeType<ConsensusType = ValidatingConsensus>,
+        I: NodeImplementation<
+            TYPES,
+            Leaf = ValidatingLeaf<TYPES>,
+            ConsensusMessage = ValidatingMessage<TYPES, I>,
+        >,
+    > HotShotType<TYPES, I> for HotShot<ValidatingConsensus, TYPES, I>
+where
+    I::Exchanges: ValidatingExchangesType<TYPES, Message<TYPES, I>>,
+    ValidatingQuorumEx<TYPES, I>: ConsensusExchange<
+        TYPES,
+        Message<TYPES, I>,
+        Proposal = ValidatingProposal<TYPES, ValidatingLeaf<TYPES>>,
+        Certificate = QuorumCertificate<TYPES, ValidatingLeaf<TYPES>>,
+        Commitment = ValidatingLeaf<TYPES>,
+    >,
+{
+    fn transactions(
+        &self,
+    ) -> &Arc<SubscribableRwLock<HashMap<Commitment<TYPES::Transaction>, TYPES::Transaction>>> {
+        &self.transactions
+    }
+
+    fn hotstuff(&self) -> &Arc<RwLock<Consensus<TYPES, I::Leaf>>> {
+        &self.hotstuff
+    }
+
+    async fn spawn_all(&self) -> HotShotHandle<TYPES, I> {
+        let shut_down = Arc::new(AtomicBool::new(false));
+        let started = Arc::new(AtomicBool::new(false));
+
+        let exchange = self.inner.exchanges.quorum_exchange();
+
+        let network_broadcast_task_handle = async_spawn(
+            tasks::network_task(
+                self.clone(),
+                shut_down.clone(),
+                TransmitType::Broadcast,
+                exchange.clone().into(),
+            )
+            .instrument(info_span!("HotShot Broadcast Task",)),
+        );
+        let network_direct_task_handle = async_spawn(
+            tasks::network_task(
+                self.clone(),
+                shut_down.clone(),
+                TransmitType::Direct,
+                exchange.clone().into(),
+            )
+            .instrument(info_span!("HotShot Direct Task",)),
+        );
+
+        async_spawn(
+            tasks::network_lookup_task(self.clone(), shut_down.clone())
+                .instrument(info_span!("HotShot network lookup task",)),
```

Comment:
nit
```suggestion
                .instrument(info_span!("HotShot Network Lookup Task",)),
```

Reply:
Also found some other places that needed capitalization. I'll update them all.